From: Guenter Roeck <linux@roeck-us.net>
To: linux-hwmon@vger.kernel.org
Cc: linux-kernel@vger.kernel.org,
	"Clemens Ladisch" <clemens@ladisch.de>,
	"Jean Delvare" <jdelvare@suse.com>,
	"Brad Campbell" <lists2009@fnarfbargle.com>,
	"Ondrej ÄŒerman" <ocerman@sda1.eu>,
	"Bernhard Gebetsberger" <bernhard.gebetsberger@gmx.at>,
	"Holger Kiehl" <Holger.Kiehl@dwd.de>,
	"Michael Larabel" <michael@phoronix.com>,
	"Jonathan McDowell" <noodles@earth.li>,
	"Ken Moffat" <zarniwhoop73@googlemail.com>,
	"Sebastian Reichel" <sebastian.reichel@collabora.com>,
	"Darren Salt" <devspam@moreofthesa.me.uk>,
	"Guenter Roeck" <linux@roeck-us.net>
Subject: [PATCH v4 5/6] hwmon: (k10temp) Don't show temperature limits on Ryzen (Zen) CPUs
Date: Wed, 22 Jan 2020 08:07:59 -0800
Message-ID: <20200122160800.12560-6-linux@roeck-us.net> (raw)
In-Reply-To: <20200122160800.12560-1-linux@roeck-us.net>

The maximum Tdie or Tctl is not published for Ryzen CPUs. What is
known, however, is that the traditional value of 70 degrees C is no
longer correct. On top of that, the limit applies to Tctl, not to Tdie.
Displaying it in either context is meaningless, confusing, and wrong.
Stop doing it.

Tested-by: Brad Campbell <lists2009@fnarfbargle.com>
Tested-by: Holger Kiehl <holger.kiehl@dwd.de>
Tested-by: Michael Larabel <michael@phoronix.com>
Tested-by: Jonathan McDowell <noodles@earth.li>
Tested-by: Ken Moffat <zarniwhoop73@googlemail.com>
Signed-off-by: Guenter Roeck <linux@roeck-us.net>
---
 drivers/hwmon/k10temp.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hwmon/k10temp.c b/drivers/hwmon/k10temp.c
index b961e12c6f58..4a470b5195ee 100644
--- a/drivers/hwmon/k10temp.c
+++ b/drivers/hwmon/k10temp.c
@@ -355,7 +355,7 @@ static umode_t k10temp_is_visible(const void *_data,
 			}
 			break;
 		case hwmon_temp_max:
-			if (channel)
+			if (channel || data->show_tdie)
 				return 0;
 			break;
 		case hwmon_temp_crit:
-- 
2.17.1
